lib/consts.ml

```diff
@@ -24,3 +24,7 @@ let wait_for_xapi_retry_delay_seconds = 4.0
 
 (** Allow no more than this many client connections. *)
 let connection_limit = 16
+
+(** Disconnect any client connection that keeps us waiting this long
+    for a read or write. *)
+let socket_timeout_seconds = 17.0
```

Comment:
Isn't 17 somewhat conservative?

Reply:
@mseri ?

Reply:
Most of our timeouts are on the order of minutes, I was wandering why 17 seconds. I don't have strong opinions. If you are all happy with it, it's fine

Reply:
@edwintorok @thomasmck @robertbreker What do you think about this timeout? Should it be longer?
I think it should be longer.

Reply:
The ticket originally said that the timeout should be 5 minutes.

Reply:
FWIW nginx uses 1 minute, but how many bytes do you read/write in each of these requests? If it is X bytes / 17s you can calculate what speed that would be, and if it turns out a modem could meet that speed then no need to increase it...